tests/unit/config/test_qtargs.py

```diff
@@ -428,6 +428,39 @@ def test_env_vars(self, monkeypatch, config_stub,
 
         assert os.environ[envvar] == expected
 
+    @pytest.mark.parametrize('config_opt, config_val, init_val, envvar, expected', [
+        ('qt.environ', {'QT_SCALE_FACTOR': '2'},
+         [None], ['QT_SCALE_FACTOR'], ['2']),
+        #Test changing an environment variable
+        ('qt.environ', {'QT_SCALE_FACTOR': '4'},
+         ['2'], ['QT_SCALE_FACTOR'], ['4']),
+        ('qt.environ', {'QT_SCALE_FACTOR': '4'},
+         ['unset'], ['QT_SCALE_FACTOR'], ['4']),
+        #Test unsetting an environment variable
+        ('qt.environ', {'QT_SCALE_FACTOR': 'None'},
+         ['3'], ['QT_SCALE_FACTOR'], ['']),
+        #Test setting multiple environment variables
+        ('qt.environ', {'QT_SCALE_FACTOR': '3', 'QT_PLUGIN_PATH': '/tmp/',
+                        'QT_NEWVAR': 'newval'},
+         [None, None, None], ['QT_SCALE_FACTOR', 'QT_PLUGIN_PATH', 'QT_NEWVAR'],
+         ['3', '/tmp/', 'newval']),
+    ])
+    def test_environ_settings(self, monkeypatch, config_stub,
+                              config_opt, config_val, init_val, envvar, expected):
+        """Test extra environment settings."""
+        for (var, val) in zip(envvar, init_val):
+            if val is not None:
+                if val == 'unset':
+                    monkeypatch.delenv(var)
+                else:
+                    monkeypatch.setenv(var, val)
+
+        config_stub.set_obj(config_opt, config_val)
```

Comment:
```suggestion
        config_stub.val.qt.environ = config_val
```

You can then also remove it from the parametrization arguments above, since it's always `qt.environ` anyways.